Engine/Module/ModuleResourceManager.cpp

```diff
@@ -91,6 +99,47 @@ update_status ModuleResourceManager::PreUpdate()
 		RefreshResourceCache();
 	}
 
+
+#if MULTITHREADING
+	if(!processing_textures_queue.Empty())
+	{
+		TextureLoadJob texture_job;
+		if(processing_textures_queue.TryPop(texture_job))
+		{
+			//Pass type variable
+			texture_job.loaded_data.texture_type = texture_job.texture_type;
+
+			if(texture_job.already_in_cache)
+			{
+				//Retrieve Texture from cache
+				texture_job.component_to_load->GetTextureFromCache(texture_job.loaded_data);
```

Comment:
Here instead of geting generting is the same but with a Init

Reply:
Well actually is what it does, but I understand that what you propose is that the component itself stores the data of the loaded_data and this way it hasnt to be passed all the time.